public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -100,11 +107,18 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
       return of({ data: [], state: LoadingState.Done });
     }
 
+    // Get linked logs datasource. Fall back to legacy loki search/trace to logs config
+    const legacyLogsDatasourceUid =
+      this.tracesToLogs?.lokiSearch !== false && this.lokiSearch === undefined
+        ? this.tracesToLogs?.datasourceUid
+        : undefined;
```

Comment:
Would be cool to extract this to util method and share with line 70-73 in QueryField.